src/Compilers/Core/Portable/Serialization/ObjectData.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+
+namespace Roslyn.Utilities
+{
+    internal class ObjectData
```

Comment:
actually, looks like ObjectData used as a key in conditional weak table, so immutable array won't work.